lib/rest/index.js

```diff
@@ -9,7 +9,11 @@ module.exports = function ({ plugins, config } = {}) {
     logger.verbose('Admin server is not configured, launch canceled');
     return;
   }
-  cfg.admin.hostname = cfg.admin.hostname || 'localhost';
+  if (cfg.admin.hostname) {
+    // eslint-disable-next-line no-console
+    console.log('Warning! use of hostname is depracated in admin section, use host instead');
+  }
+  cfg.admin.host = cfg.admin.hostname || cfg.admin.host || 'localhost';
```

Comment:
Let's remove the default value and put it in the JSON Schema

```suggestion
  cfg.admin.host = cfg.admin.hostname || cfg.admin.host;
```

Reply:
If we remove `localhost` default value from here, then `28 tests` are breaking.
- One of them is https://github.com/ExpressGateway/express-gateway/blob/8d4e9646e795538ee2abf272b190cabff0ae7568/test/rest-api/api-endpoint.test.js#L24
- I think, in these tests, the `yaml config` is made up directly from object defined there and `host` is missing there, so it relies on this default value in the `lib/rest/index.js` file and not on the JSON schema default value.
___
If that's the case, then we've two options:
- Add `host` in the config objects for all the failing tests
-  or Leave this default value as it is here

Please let me know if I missed something

Reply:
@ravikp7 The reason is a wrong JSON Schema. It can be tricky sometimes to modify it correctly so I went ahead and modified it for you. See the changes at ca50b189fb7fa1f961bbf0021194d52a6cdd9e47

Basically what is needed is to declare both property in the schema, and then use the `oneOf` feature to specify that only one of the two is required.

I've also modified the order — as we want to give the new property `host` precedence over `hostname`